### ISSUE TITLE
Using device orientation on iOS5 causes geolocation lookups for compass calibration

### ISSUE THREAD

**Issue by alexgibson** [scottjehl/iOS-Orientationchange-Fix#1]
I've already posted this on the original Gist, but replicated here on the repo as well.

I fear this could be quite dangerous, especially if adopted by a large project that is widely used?

---

This does work great - very clever solution! My only reservation is that iOS5 uses geolocation for compass calibration the whole time the browser window is open. I'm not sure what effect this has on battery life, but might be worth consideration?

You can enable the status bar icon for this in:

Settings -> Location Services -> System Services -> Status Bar Icon set to On

Edit - perhaps an alternative event could be to use devicemotion instead of deviceorientation? This would mean it does not require a gyroscope (only iPhone 4+), and could be done using just raw accelerometer events.

**Comment by scottjehl** [comment]
Interesting observations, Alex! Thanks.

If you can get it to work with deviceorientation, I suppose that'd be better here.  Please send a pull if you find anything.

**Comment by scottjehl** [comment]
Oh sorry I read that wrong. So I thought orientation would be the better option here, would it not? Doesn't devicemotion fire more frequently?

**Comment by alexgibson** [comment]
Yes devicemotion does fire more frequently, but that is a lot less costly than making constant lookups to a location service? If my concerns here are correct, this bug fix (while a very clever idea) could cause much more serious problems for users that the one it tries to fix? i.e. draining battery, maybe data usage? It would need some more investigating...

Maybe I am being over-cautious here, but that's not necessarily a bad thing - these hidden system services run silently in the background (unless set in preferences as shown above). Many people may never realise this is happening before using it on their sites.

Perhaps if you could disable webkitCompassHeading somehow, or check for orientation using devicemotion it would be a better method? It could also then work on devices that don't have a gyroscope?

**Comment by scottjehl** [comment]
Huh, I guess I was unaware that the event needed a location service for any of this information - all the documentation I can find on it says it uses the iPhone's internal accelerometer.

Please let me know what you find! Thanks :)

**Comment by alexgibson** [comment]
The webkitCompassHeading property was added to iOS5, but there is little info on how it works (internally) in the Safari Reference docs:

http://developer.apple.com/library/safari/#documentation/SafariDOMAdditions/Reference/DeviceOrientationEventClassRef/DeviceOrientationEvent/DeviceOrientationEvent.html#//apple_ref/doc/uid/TP40010526

I only stumbled upon the Compass Calibration setting in location services by mistake. But if you enable the location services status bar for this setting, it does indeed show a purple arrow in the status bar the whole time the demo page is open, suggesting to me that it continuously polls in the background in order to get an accurate due north heading. This remains active as long as the page is open in the browser.
